Sources/ArcGISToolkit/Components/OfflineMapAreasView/OfflineMapAreasView.swift

```diff
@@ -0,0 +1,109 @@
+    @StateObject private var mapViewModel: MapViewModel
+    
+    /// The action to dismiss the view.
+    @Environment(\.dismiss) private var dismiss: DismissAction
+    
+    public init(map: Map) {
+        _mapViewModel = StateObject(wrappedValue: MapViewModel(map: map))
+    }
+    
+    public var body: some View {
+        NavigationStack {
+            Form {
+                if mapViewModel.hasPreplannedMapAreas {
+                    Section(header: Text("Preplanned Map Areas").bold()) {
+                        preplannedMapAreas
+                    }
+                    .textCase(nil)
+               }
+            }
+            .task {
+                await mapViewModel.makePreplannedOfflineMapModels()
+            }
+            .toolbar {
+                ToolbarItem(placement: .confirmationAction) {
+                    Button("Done") { dismiss() }
+                }
+            }
+            .navigationTitle("Offline Maps")
+            .navigationBarTitleDisplayMode(.inline)
+        }
+    }
+    
+    @ViewBuilder private var preplannedMapAreas: some View {
+        switch mapViewModel.preplannedMapModels {
+        case .success(let models):
+            List(models) { preplannedMapModel in
+                PreplannedListItemView(
+                    preplannedMapModel: preplannedMapModel
+                )
+            }
+        case .failure(let error):
+            Text(error.localizedDescription)
+        case .none:
+            ProgressView()
+                .frame(maxWidth: .infinity)
+        }
+    }
+}
+
+public extension OfflineMapAreasView {
+    /// The model class for the offline map areas view.
+    @MainActor
+    class MapViewModel: ObservableObject {
+        /// The online map.
+        private let onlineMap: Map
+        
+        /// The offline map task.
+        private let offlineMapTask: OfflineMapTask
+        
+        /// The preplanned offline map information.
+        @Published private(set) var preplannedMapModels: Result<[PreplannedMapModel], Error>?
+        
+        /// A Boolean value indicating whether the map has preplanned map areas.
+        @Published var hasPreplannedMapAreas: Bool = false
+        
+        init(map: Map) {
+            self.onlineMap = map
+            
+            offlineMapTask = OfflineMapTask(onlineMap: onlineMap)
+        }
+        
+        /// Gets the preplanned map areas from the offline map task and creates the
+        /// offline map models.
+        func makePreplannedOfflineMapModels() async {
+            self.preplannedMapModels = await Result {
+                try await offlineMapTask.preplannedMapAreas
+                    .sorted(using: KeyPathComparator(\.portalItem.title))
+                    .compactMap {
+                        PreplannedMapModel(
+                            preplannedMapArea: $0
+                        )
+                    }
+            }
+            if let models = try? preplannedMapModels?.get() {
```

Comment:
05/14: Let's chat about ways to handle this error. I think showing a red Text with the error description is fine. Maybe grab a screenshot for the demo with network link conditioner. 🤔